src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -207,15 +207,15 @@ private RestClientMethod BuildMethod(Operation operation, string clientName, IRe
                     switch (requestParameter.Schema)
                     {
                         case ConstantSchema constant:
-                            constantOrParameter = ParseConstant(constant);
+                            constantOrParameter = BuilderHelpers.ParseConstant(constant, _typeFactory);
                             valueSchema = constant.ValueType;
                             break;
                         default:
                             constantOrParameter = BuildParameter(requestParameter);
                             break;
                     }
 
-                    if (!constantOrParameter.IsConstant)
+                    if (!constantOrParameter.IsConstant && requestParameter.Flattened != true)
```

Comment:
This makes sure we don't put any of the flattened parameters into the method definition.

Reply:
Why do we need to do it? don't we have them as method parameterS?

Reply:
No. `requestParameter.Flattened` are the parameters (original) that are of the model type. The actual parameters we want to use are of type VirtualParameters. This entire request building process could be refined if we used signatureParameters, because that filtering work was already done for us. However, currently, we're creating a constant parameter, and signatureParameters won't have constants in it. Plus, that work to use signatureParameters it outside the scope of these changes.